src/Struct/FindologicProduct.php

```diff
@@ -350,17 +357,15 @@ protected function getTranslatedDomainBaseUrl(): ?string
         $salesChannel = $this->salesChannelContext->getSalesChannel();
         $domainCollection = $salesChannel->getDomains();
 
-        /** @var SalesChannelDomainEntity|null $domainEntity */
-        $domainEntity = $this->getTranslatedEntity($domainCollection);
+        $domainEntities = $this->getTranslatedEntities($domainCollection);
 
-        return $domainEntity ? rtrim($domainEntity->getUrl(), '/') : null;
+        return $domainEntities->first() ? rtrim($domainEntities->first()->getUrl(), '/') : null;
```

Comment:
Is it also safe to call first() here?

Reply:
The same was done before just in a different place. If there is no domain, then we just return `null` here and similar to if we have no SEO URL, we fallback to the default product URL.